### PR TITLE
[Fix] fix torch allocator resouce releasing

### PR DESCRIPTION
**Pull request by AllentDan**
Reproduce codes in master branch:
```python
from mmdeploy.backend.tensorrt.wrapper import TRTWrapper
trt_model = TRTWrapper(
    '/path_to/cls/resnet/end2end.engine',
    ['output'])
input = torch.rand(1, 3, 224, 224).cuda()
output_trt = trt_model.forward({'input': input})
print(output_trt)
```
Error might triggered:
```
/mmdeploy/utils/logging.py(28): get_logger
Exception caught in deallocate(): TypeError: 'NoneType' object is not callable
```
or
```
codes/mmdeploy/mmdeploy/backend/tensorrt/torch_allocator.py(61): deallocate

[ERROR] Exception caught in deallocate(): AttributeError: 'NoneType' object has no attribute 'caching_allocator_delete'
```

### CODE REVIEW

File: mmdeploy/backend/tensorrt/torch_allocator.py [hunk 1]
```diff
@@ -53,11 +53,10 @@ def deallocate(self: trt.IGpuAllocator, memory: int) -> bool:
         Returns:
             bool: deallocate success.
         """
-        logger = get_root_logger()
```
Comment (lvhan028) [inline mmdeploy/backend/tensorrt/torch_allocator.py:56]: Can we still keep them if mmengine is installed?

Comment (AllentDan) [reply mmdeploy/backend/tensorrt/torch_allocator.py:56]: Yes. We can also modify the codes of [get_logger()](https://github.com/open-mmlab/mmdeploy/blob/master/mmdeploy/utils/logging.py).

**Comment by grimoire** [comment]
Unable to reproduce the error.
Why would `logging` and `torch.cuda` be None?

**Comment by AllentDan** [comment]
> Unable to reproduce the error. Why would `logging` and `torch.cuda` be None?

Don't know either. Just happened at the end of the program. Would it be possible that torch.cuda and logging are released at that time.

**Comment by grimoire** [comment]
More investigations are required.

**Review by grimoire — APPROVED** [review]

**Comment by grimoire** [comment]
Could report to TensorRT team.

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**